arangod/VocBase/LogicalDataSource.cpp

```diff
@@ -53,7 +53,15 @@ std::string ensureGuid(std::string&& guid, TRI_voc_cid_t id, TRI_voc_cid_t planI
   // id numbers can also not conflict, first character is always 'h'
   if (arangodb::ServerState::instance()->isCoordinator() ||
       arangodb::ServerState::instance()->isDBServer()) {
-    TRI_ASSERT(planId);
+     // this was seen to happen on coordinator when agency returned '0' for 'id'
+    // and no 'planId' was provided in the definition
+    if (!planId) {
```

Comment:
wait this should be handled somwhere else ? If the agency returns 0 we should not get here

Reply:
alas I saw it with my own eyes, spent quite a while tracking the failure to this specific assert since the debugger didn't report the correct line

Reply:
```
> I think you should handle this error in ensureId
as far as I understand it's valid for 'id' to be 0 in some cases where the DataSource is a temporary or uninitialized object, so I am unsure if ensureId(...) is the correct place for this check
it's actually a planId==0 that is a problem that is why the assert was specifically for planId there before

Reply:
I think the ID should never be 0. What I mean is that the code that generates the plan ID will call `ClusterInfo::uniqueId()`. If this returns 0 it is an error, maybe we are not handling the error appropriately

Reply:
but I will leave it up to you

Reply:
```
> I think the ID should never be 0. 
that is my understanding too, I am uncertain under what conditions it is allowed to be 0 since 0 is sometimes hardcoded and nothing fails
> What I mean is that the code which generates the plan ID will call ClusterInfo::instance()->uniqueId().
the code that generates plan ID just reuses the value from '_id', and for the cluster case '_id' is taken from uniqueId(), so the question is what is the error, that '_id' == 0 or that '_planId' is taken from '_id'?

Reply:
oh never mind there is a TRI_ASSERT(_id); on line 189